modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetHandler.java

```diff
@@ -1030,7 +1031,7 @@ public void exception(NHttpClientConnection conn, Exception ex) {
             } else {
                 log.error("Unexpected error.");
             }
-            if (targetConfiguration.isCorrelationLoggingEnabled()) {
+            if (PassThroughCorrelationConfigDataHolder.isEnable()) {
```

Comment:
Since we have updated the `isCorrelationLoggingEnabled()` method in `BaseConfiguration` class to return the `PassThroughCorrelationConfigDataHolder` value, do we need to change all these places as well?

Reply:
As this layer is performance oriented, @sachithKay suggested instead of calling two function calls, we should directly check the dataholder.